awx/api/views/__init__.py

```diff
@@ -408,6 +409,54 @@ def is_valid_relation(self, parent, sub, created=False):
         return None
 
 
+class InstanceHealthCheck(GenericAPIView):
+
+    name = _('Instance Health Check')
+    model = models.Instance
+    serializer_class = serializers.InstanceHealthCheckSerializer
+    permission_classes = (IsSystemAdminOrAuditor,)
+
+    def get(self, request, *args, **kwargs):
+        obj = self.get_object()
+        data = self.get_serializer(data=request.data).to_representation(obj)
+        return Response(data, status=status.HTTP_200_OK)
+
+    def post(self, request, *args, **kwargs):
+        obj = self.get_object()
+
+        if obj.node_type == 'execution':
+            from awx.main.tasks import execution_node_health_check
+
+            runner_data = execution_node_health_check(obj.hostname)
```

Comment:
I'm wondering if this request is ever going to timeout waiting for this to complete. I guess `ansible-runner worker --worker-info` doesn't require pulling an image, so it should be pretty quick to respond. So, probably not an issue.

Reply:
For the _execution node_ case, the maximum time it could take is practically 20 seconds, because _that method_ enforces timeouts, and will return errors if timeout happens.

But you reminded me that I didn't handle the timeout case for control nodes... which should be _exceedingly_ rare but I will still add something really quick here.